wp_localization/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+use fluent_bundle::FluentValue;
+use fluent_templates::Loader;
+use std::{collections::HashMap, fmt::Debug, fmt::Display};
+use strum_macros::{EnumIter, IntoStaticStr};
+
+fluent_templates::static_loader! {
+    static LOCALES = {
+        locales: "./localization",
+        fallback_language: "en-US"
+    };
+}
+
+#[wp_localization_macro::wp_messages]
+pub struct WpMessages {}
+
+#[derive(Debug)]
+pub struct MessageBundle {
+    // Keep the fields private to avoid clients from looking up messages that do not exist.
+    key: &'static str,
+    args: Option<HashMap<&'static str, String>>,
+}
+
+impl MessageBundle {
+    // Keep this private to avoid clients from looking up messages that do not exist.
+    fn new(key: &'static str, args: Option<HashMap<&'static str, String>>) -> Self {
+        Self { key, args }
+    }
+
+    pub fn key(&self) -> &'static str {
+        self.key
+    }
+
+    pub fn args(&self) -> Option<&HashMap<&'static str, String>> {
+        self.args.as_ref()
+    }
+
+    pub fn localize(&self, locale: Option<WpLocale>) -> String {
+        LOCALES.lookup_complete(
+            &locale.unwrap_or_default().as_language_id(),
+            self.key,
+            self.args
+                .as_ref()
+                .map(|h| {
+                    h.iter()
+                        .map(|(k, v)| ((*k).into(), FluentValue::from(v)))
+                        .collect()
+                })
+                .as_ref(),
+        )
+    }
+}
+
+impl Display for MessageBundle {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.localize(None))
+    }
+}
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    Default,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    EnumIter,
+    IntoStaticStr,
+    uniffi::Enum,
+)]
+pub enum WpLocale {
+    #[default]
+    #[strum(serialize = "en-US")]
+    EnUS,
+    #[strum(serialize = "tr-TR")]
+    TrTR,
+}
+
+impl WpLocale {
+    pub fn as_language_id(&self) -> unic_langid::LanguageIdentifier {
+        Into::<&str>::into(self).parse().expect(
+            "All locales are unit tested to ensure they can be converted to LanguageIdentifier",
+        )
+    }
```

Comment:
I think we should be able to use the `langid!` macro to avoid parsing the same string repeatedly at runtime. For example, generating the `WpLocale` type in the build script and adding a `fn as_language_id() -> &'static LanguageIdentifier`.

Reply:
I missed the fact that `langid!` can be used to create constants. Good catch, I'll address this tomorrow.